### ISSUE TITLE
[RFC] v0.1 Release Tracker

### ISSUE THREAD

**Issue by SamitHuang** [verl-project/verl-omni#47]
Cut verl-omni v0.1 as the first tagged release after going public. The focus is stability, alignment with upstreams, and solid reference recipes for diffusion models.

P1 items are listed in #97 

### Training Stability

* Efficient or stable algorithms
  - [x] GRPO-guard algorithm implementation @zhtmike #48 
  - [x] MixGRPO algorithm implementation @SamitHuang  #58 
  - [x]  DiffusionNFT @AndyZhou952 #106 

### Efficiency

* Upstream framework upgrade
  - [x] VeRL alignment (main or v0.8) @SamitHuang @AndyZhou952   https://github.com/verl-project/verl-omni/pull/119
  - [x] vLLM-omni alignment (v0.20.2, v0.22)  @knlnguyen1802 https://github.com/vllm-project/vllm-omni/pull/3379 
    - [x] drop master port patch code as https://github.com/vllm-project/vllm-omni/pull/3803 merged @zhtmike 
   
* Profiling support
  - [x] FlopsCounter for DiTs #124  #128 
  - [x] Adopt the unified profiling framework in verl (verl.utils.profiler) and give worked examples @zhtmike #46 
  - [x] Add profiling docs in developer guide @zhtmike #46 

* Peformance reference
  - [x] Improve QwenImage FlowGRPO reference report by covering MFU, throughput, step breakdown, and wandb log @AndyZhou952

- [x] Diffusion Rollout Correction: https://github.com/verl-project/verl-omni/pull/93

* Multi-node Parallel Training
  - [x] FlowGRPO #118 

* System optimizations
  - [x] Rollout: 
    - [x] adopt step-wise continuous batching @knlnguyen1802   https://github.com/vllm-project/vllm-omni/pull/2729  @https://github.com/vllm-project/vllm-omni/pull/3639
  - [x] Reward: asynchronized reward computation
  - [x] Trainer: 
    - [x] support SP on DiffusersFSDPTrainer to reduce memory cost @zhtmike #59 

### CI tests and Usability
- [x] **Setup GPU smoking tests** @zhtmike #45
- [x] Remove reward worker patch code  @zhtmike  #56 
- [ ]  Standard convergence test setup: QwenImage/Qwen3omni + DPO/GRPO + LoRA/Full-param long-run training (local tests)

### New Models and Features
- [x] Qwen-Image
  - [x] **Full-model** FlowGRPO post-training [P0] @AndyZhou952  #55 
  - [x] DPO post-training @wtomin #95 
- [ ] Qwen3-omni(thinker) GSPO training
  - [x] LoRA fine-tuning @qinganrice #113 
  - [ ] Integrate megatron-core trainer for e2e training @hbhflw2000 
 - [x] BAGEL
   - [x] LoRA FlowGRPO fine-tuning @princepride  @zhtmike  #66 

### Hardware support
- [ ] NPU integration @panshaowu  https://github.com/verl-project/verl-omni/issues/91

### Documentations
- [x] Add design docs

### CC List.

@wuxibin89 @zhtmike @princepride @panshaowu @hsliuustc0106  @AndyZhou952 

Related issue: Q2 roadmap https://github.com/verl-project/verl/issues/5755

**Comment by zhtmike** [comment]
Plus, we need to support LoRA checkpoint saving in engine worker; it is not supported in verl main branch yet.

**Comment by wtomin** [comment]
Please assign DPO post-training + qwen-image task to me. @SamitHuang

**Comment by zhtmike** [comment]
Since diffusers 0.38 released, I can help to integrate SP for diffusers training.
It is useful for full model RL to address the memory issues

**Comment by hbhflw2000** [comment]
I’m interested in helping with the “Integrate megatron-core trainer for e2e training”  for Qwen3-Omni.

I’m currently working on Qwen3-Omni support in Megatron-Bridge, including convert / infer / train paths. I think this can serve as the Megatron/MBridge backend for full-model / full-weight post-training in verl-omni.

**Comment by uygnef** [comment]
Hi @SamitHuang I’m interested in  Fully-model post-training for Qwen3-omni(thinker) GSPO training.

**Comment by SamitHuang** [comment]
> Hi [@SamitHuang](https://github.com/SamitHuang) I’m interested in Fully-model post-training for Qwen3-omni(thinker) GSPO training.

Thanks for the interest. Assigned. It may reply on the megatron trainer.  You can join the wechat channel to discuss with @hbhflw2000 and @princepride  as well

**Comment by niehen6174** [comment]
Hi @SamitHuang @zhtmike  Could you assign the LTX 2.3 task to me?
Building upon your existing work, I have successfully implemented Flow Group training for both the LTX 2.3 LoRA and full models. Using PickScore and VideoAlign for rewards, the performance metrics have shown a consistent upward trend. However, I have currently made some modifications to the rollout mechanism—specifically, I am not utilizing the `vllm-omni` rollout implementation. Once I have finished adapting and integrating these changes, I should be ready to submit the code.

**Comment by SamitHuang** [comment]
> Hi [@SamitHuang](https://github.com/SamitHuang) [@zhtmike](https://github.com/zhtmike) Could you assign the LTX 2.3 task to me? Building upon your existing work, I have successfully implemented Flow Group training for both the LTX 2.3 LoRA and full models. Using PickScore and VideoAlign for rewards, the performance metrics have shown a consistent upward trend. However, I have currently made some modifications to the rollout mechanism—specifically, I am not utilizing the `vllm-omni` rollout implementation. Once I have finished adapting and integrating these changes, I should be ready to submit the code.

thanks. assigned

**Comment by yuekaizhang** [comment]
@SamitHuang  Hi, I was wondering if you plan to support speech generation post-training. If so, I’m interested in supporting qwen3-tts in verl-omni.

**Comment by wuli666** [comment]
Hi @SamitHuang , I'd like to take the Support multi-reward computation  task. I've worked on something similar before. Could you assign it to me?

**Comment by SamitHuang** [comment]
> [@SamitHuang](https://github.com/SamitHuang) Hi, I was wondering if you plan to support speech generation post-training. If so, I’m interested in supporting qwen3-tts in verl-omni.

yes, assigned. It'd be good to create a RFC issue to discuss the design and key changes to support audio generation model family (e.g. audio reward model, audio encoder/decoder)

**Comment by SamitHuang** [comment]
> Hi [@SamitHuang](https://github.com/SamitHuang) , I'd like to take the Support multi-reward computation task. I've worked on something similar before. Could you assign it to me?

Sure, please create a RFC to discuss the architecture design, implementation plan, and verification task settings (ocr + pickscore reward?)

**Comment by wtomin** [comment]
Offline DPO traning PR is ready: https://github.com/verl-project/verl-omni/pull/95/ @SamitHuang
